toolchain/sem_ir/inst_namer.cpp

```diff
@@ -245,16 +249,28 @@ auto InstNamer::Namespace::AllocateName(const InstNamer& inst_namer,
 
   // Append location information to try to disambiguate.
   // TODO: Consider handling inst_id cases.
-  if (loc_id.is_node_id()) {
-    const auto& tree = inst_namer.sem_ir_->parse_tree();
-    auto token = tree.node_token(loc_id.node_id());
-    llvm::raw_string_ostream(name)
-        << ".loc" << tree.tokens().GetLineNumber(token);
-    add_name();
-
-    llvm::raw_string_ostream(name)
-        << "_" << tree.tokens().GetColumnNumber(token);
-    add_name();
+  if (LocId* loc_id = std::get_if<LocId>(&loc_id_or_fingerprint)) {
+    if (loc_id->is_node_id()) {
+      const auto& tree = inst_namer.sem_ir_->parse_tree();
+      auto token = tree.node_token(loc_id->node_id());
+      llvm::raw_string_ostream(name)
+          << ".loc" << tree.tokens().GetLineNumber(token);
+      add_name();
+
+      llvm::raw_string_ostream(name)
+          << "_" << tree.tokens().GetColumnNumber(token);
+      add_name();
+    }
+  } else {
+    uint64_t fingerprint = std::get<uint64_t>(loc_id_or_fingerprint);
+    llvm::raw_string_ostream out(name);
+    out << ".";
```

Comment:
What would you think of a prefix for fingerprints like the `.loc` for location ids? Like `.fin` or `.fp` or something?

Reply:
Thanks, yes I like the 3 digits a lot more.